src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1521,7 +1521,7 @@ function ($a) {
                                 ->addDefaultsIfNotSet()
                                 ->children()
                                     ->enumNode('default_middleware')
-                                        ->values([true, false, 'allow_no_handlers'])
+                                        ->values([true, false, 'allow_no_handlers', 'allow_no_senders'])
```

Comment:
Following the comment from @HeahDude, I think it would be nice to turn this default_middleware option into an array node:
```php
->arrayNode('default_middleware')
    ->canBeDisabled()
    ->children()
        ->booleanNode('allow_no_handlers')->defaultFalse()->end()
        ->booleanNode('allow_no_senders')->defaultTrue()->end()
```
With an appropriate `beforeNormalization()` closure to turn the old configs into the new one.
WDYT?

Reply:
But then there would be conflict between `allow_no_handlers` and `enabled: true` that would need proper handling.

Reply:
which conflict? "enabled" would be about "default_middleware", not about allow_no_handlers

Reply:
it requires `enable: true` yes. enabled/disabled applies to a whole subnode, there's nothing specific to this node here. If you define this, then of course `allow_no_handlers` is going to be ignored:
```
default_middleware:
  enabled: false
  allow_no_handlers: true
```